### PR TITLE
[opt](cloud) Make get tablet stats and update delete bitmap update lock be able to be in different fdb txns

### PR DESCRIPTION
**Pull request by bobhan1**
### What problem does this PR solve?

https://github.com/apache/doris/pull/37670 let the meta service return the tablet compaction stats along with the `getDeleteBitmapUpdateLockResponse` to FE to let the BE know whether it should `sync_rowsets()` due to successful compaction on other BEs on the same tablet. That PR makes the process of reading tablets' stats and writing the delete bitmap update lock KV in one fdb txn to achieve the atomic sematic. However, when a load involves a large number of tablets, the process of reading tablets' stats may take longer than fdb txn's 5 seconds limitation and cause `TXN_TOO_OLD` error.
This PR re-arrange the process so that the read of tablet stats can be not necessarily in the same fdb txn with the txn which update the lock_info.lock_id. In detail, we do as the following:
1. gain the delete bitmap update lock in MS (write delete bitmap update lock KV)
2. read tablets' stats to get the compaction counts.
3. check if the delete bitmap update lock is still held by the current load.

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [x] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

**Review by zhannngchen — DISMISSED** [review]
LGTM

File: cloud/src/meta-service/meta_service.cpp [hunk 1]
```diff
@@ -2193,6 +2169,73 @@ void MetaServiceImpl::get_delete_bitmap_update_lock(google::protobuf::RpcControl
...
+            return;
+        }
+
+        int64_t total_retry = 0;
+        for (const auto& tablet_index : request->tablet_indexes()) {
+            TabletIndexPB idx(tablet_index);
+            TabletStatsPB tablet_stat;
+            int64_t retry = 0;
+            internal_get_tablet_stats(code, msg, txn.get(), instance_id, idx, tablet_stat, false);
+            while (retry < 3 && total_retry < 20 && code == MetaServiceCode::KV_TXN_TOO_OLD) {
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service.cpp:2198]: this is a point-get, it is unlikely to get `TOO_OLD`, this retry does not make sense.

File: cloud/src/meta-service/meta_service.cpp [hunk 2]
```diff
@@ -2193,6 +2169,73 @@ void MetaServiceImpl::get_delete_bitmap_update_lock(google::protobuf::RpcControl
...
+            response->add_cumulative_compaction_cnts(tablet_stat.cumulative_compaction_cnt());
+            response->add_cumulative_points(tablet_stat.cumulative_point());
+        }
+
+        if (!check_delete_bitmap_lock(code, msg, ss, txn, instance_id, table_id, request->lock_id(),
+                                      request->initiator())) {
+            LOG(WARNING) << "failed to check delete bitmap lock after get tablet stats, table_id="
+                         << table_id << " request lock_id=" << request->lock_id()
+                         << " request initiator=" << request->initiator() << " msg " << msg;
+            return;
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service.cpp:2236]: redundant `return`

File: cloud/src/meta-service/meta_service.cpp [hunk 3]
```diff
@@ -2193,6 +2169,73 @@ void MetaServiceImpl::get_delete_bitmap_update_lock(google::protobuf::RpcControl
...
+            msg = "failed to init txn";
+            return;
+        }
+
+        int64_t total_retry = 0;
+        for (const auto& tablet_index : request->tablet_indexes()) {
+            TabletIndexPB idx(tablet_index);
+            TabletStatsPB tablet_stat;
+            int64_t retry = 0;
+            internal_get_tablet_stats(code, msg, txn.get(), instance_id, idx, tablet_stat, false);
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service.cpp:2197]: read stats_tablet_key

**Comment by bobhan1** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Review by gavinchou — APPROVED** [review]

**Pull request merged**